util/dvsim/LintParser.py

```diff
@@ -0,0 +1,105 @@
+            'warnings': 'Tool Warnings',
+            'errors': 'Tool Errors',
+            'lint_infos': 'Lint Infos',
+            'lint_warnings': 'Lint Warnings',
+            'lint_errors': 'Lint Errors',
+        }
+        self.severities = {
+            'warnings': 'warning',
+            'errors': 'error',
+            'lint_infos': 'info',
+            'lint_warnings': 'warning',
+            'lint_errors': 'error',
+        }
+
+
+    def extract_messages(self, log_content: str, patterns: List[str]) -> None:
+        """
+        This extracts messages from the string buffer log_content.
+        The argument patterns needs to be a list of tuples with
+        (<error_severity>, <pattern_to_match_for>).
+        """
+        for bucket, pattern in patterns:
+            self.buckets[bucket] += \
+                re.findall(pattern, log_content, flags=re.MULTILINE)
+
+    def get_results(self, args: Dict[Path, List[Tuple]]) -> Dict[str, int]:
+        """
+        Parse report and corresponding logfiles and extract error, warning
+        and info messages for each IP present in the result folder
+        """
+
+        log_file_contents = []
+        # Open all log files
+        for path, patterns in args.items():
+            try:
+                with path.open() as f:
+                    log_content = f.read()
+            except IOError as err:
+                self.buckets['errors'] += \
+                    ["IOError: %s" % err]
+            else:
+                # Go parse the logs
+                self.extract_messages(log_content, patterns)
+
+        # If there are no errors or warnings, add the "fusesoc-error" field to
+        # "errors" (which will be reported as tooling errors). Remove the
+        # "fusesoc-error" field either way.
+        num_messages = {
+            'info': 0,
+            'warning': 0,
+            'error': 0
+        }
+        for key, sev in self.severities.items():
+            num_messages[sev] += len(self.buckets[key])
+        if num_messages['error'] == 0 and num_messages['warning'] == 0:
+            self.buckets['errors'] = self.buckets['fusesoc-error']
+        del self.buckets['fusesoc-error']
+
+        return num_messages
+
+    def write_results_as_hjson(self, outpath: Path) -> None:
+        with outpath.open("w") as results_file:
+            # Construct results dict for Hjson file.
+            results_dict = {'buckets': {}}
+            for key, label in self.labels.items():
+                results_dict['buckets'].update({
+                    key: {
+                        'label': label,
+                        'severity': self.severities[key],
+                        'messages': self.buckets[key]
+                    }
+                })
+            print(results_dict)
+            hjson.dump(results_dict,
+                       results_file,
+                       ensure_ascii=False,
+                       for_json=True,
+                       use_decimal=True)
```

Comment:
Should be abstracted to a common `Parser` class that individual flows/tool classes extend from.

Reply:
Yes, let's do that as a follow up.

Reply:
Made an issue for this here: #9079